jit.js

```diff
@@ -139,6 +139,185 @@ to single-step.
             'atEnd', '==', 'class', 'blockCopy:', 'value', 'value:', 'do:', 'new', 'new:', 'x', 'y'];
     },
 },
+'peephole', {
+    getPush: function (byte) {
+        switch (byte & 0xF8) {
+            case 0x00: case 0x08:
+                return "inst[" + (byte & 0x0F) + "]";
+            case 0x10: case 0x18:
+                return "temp[" + (6 + (byte & 0xF)) + "]";
+            case 0x20: case 0x28: case 0x30: case 0x38:
+                return "lit[" + (1 + (byte & 0x1F)) + "]";
+            case 0x40: case 0x48: case 0x50: case 0x58:
+                return "lit[" + (1 + (byte & 0x1F)) + "].pointers[1]";
+            case 0x70:
+                return this.simplePush["p" + byte.toString(16)];
+        }
+    },
+    simplePush: {
+        p70: "rcvr",
+        p71: "vm.trueObj",
+        p72: "vm.falseObj",
+        p73: "vm.nilObj",
+        p74: -1,
+        p75: 0,
+        p76: 1,
+        p77: 2
+    },
+    generateStartOfNumericOp: function(bytes){
+        var push1 = this.getPush(bytes[0]);
+        var push2 = this.getPush(bytes[1]);
+
+        var op = {
+            nb0: "vm.primHandler.signed32BitIntegerFor(push1 + push2)",
+            nb1: "vm.primHandler.signed32BitIntegerFor(push1 - push2)",
+            nb2: "push1 < push2 ? vm.trueObj : vm.falseObj",
+            nb3: "push1 > push2 ? vm.trueObj : vm.falseObj",
+            nb4: "push1 <= push2 ? vm.trueObj : vm.falseObj",
+            nb5: "push1 >= push2 ? vm.trueObj : vm.falseObj",
+            nb6: "push1 === push2 ? vm.trueObj : vm.falseObj",
+            nb7: "push1 !== push2 ? vm.trueObj : vm.falseObj"
+        };
+
+        var operation = op["n" + bytes[2].toString(16)].replace("push1", push1).replace("push2", push2);
```

Comment:
Same here: don't use a dictionary, use an Array. Code like `op["n" + bytes[2].toString(16)]` is plain ugly compared to e.g. `op[bytes[2] & 7]`.